### DISCUSSION TITLE
Set a custom response status code

### DISCUSSION THREAD

**Discussion by chrisnoden** [nuxt/nuxt#17283] [Questions]
Hi

I am using Storyblok to define the pages in the majority of my site. I use the `[slug].vue` feature of Nuxt v3 to define the main page and then search for the content using the slug (route.path).

If the slug is not found on Storyblok I want to return a 404 response complete with custom 404 page. In this use case the `pages/404.vue` solution does not work.

I want to return the 404 status code (so SEO understands this page does not exist) and show a custom 404 page response for the humans.

I tried:
```
const error = new Error();
error.statusCode = 404;
error.statusMessage = 'Not Found';

throw error;
```

This just seems to upset the current build of Nuxt.

My plan then was to use v-if/v-else conditions to show either the page content or an error component, but I can't seem to get things to play without Nuxt/Vite throwing a wobbly.

**Comment by Lyokolux** [comment]
> show a custom 404 page response

You can create an `error.vue` component at the base of your project. Then you can catch the error and use `v-if` in case of 404-error.

Here is my (very) basic `~/error.vue`:
```vue
<script setup lang="ts">
type Props = {
  error: {
    url: string,
    statusCode: string,
    statusMessage: string,
    message: string,
    description: string,
    data: any
  }
}

const props = withDefaults(defineProps<Props>(), {})
</script>


<template>
  <h1>Got you {{ error.statusCode }}</h1>

  <p>{{ error.message }}</p>

  <p>{{ error.description }}</p>

  <NuxtLink to="/">Go Back Home</NuxtLink>
</template>

```

>This just seems to upset the current build of Nuxt.

Did you tried it in a defineEventHandler function ?
You can use the response object:
```typescript
export default defineEventHandler(async ({ res }) => {
  res.statusCode = 404
  res.statusMessage = 'Not Found';
  res.end() # send the message
})
```

**Reply by chrisnoden** [reply]
Thanks for helping. As mentioned, the `v-if` + `v-else` approach was already tried. The challenge is sending the 404 status and delivering the vue template. I can do one but not both.

I don't know how to define a server route/middleware that will send the 404 status AND send a vue template. That isn't covered in the nuxt/nitro/h3 docs that I can see.
